addons/payment_alipay/models/payment_transaction.py

```diff
@@ -3,132 +3,119 @@
 
 import logging
 
-from hashlib import md5  # TODO ARJ unused import
 from werkzeug import urls
 
-from odoo import api, fields, models, _  # TODO ARJ unused import with fields
+from odoo import api, models, _
 from odoo.tools.float_utils import float_compare
 from odoo.addons.payment_alipay.controllers.main import AlipayController
-from odoo.addons.payment.models.payment_acquirer import ValidationError  # TODO ARJ this is old stuff; import from odoo.exceptions now
+from odoo.exceptions import ValidationError
 
 _logger = logging.getLogger(__name__)
 
 
 class PaymentTransaction(models.Model):
     _inherit = 'payment.transaction'
 
-    # TODO ARJ CRUD methods come first -> create, write, _alipay_check_configuration
-    def _check_alipay_configuration(self, vals):
-        acquirer_id = int(vals.get('acquirer_id'))  # TODO ARJ do int(vals.get('...', 0)) to have the CRUD method raise a ValidationError instead of yours raising a TypeError if a val is missing  # TODO ARJ same below
-        acquirer = self.env['payment.acquirer'].sudo().browse(acquirer_id)  # TODO ARJ sudo shoudln't be needed here  # TODO ARJ .exists()
-        if acquirer and acquirer.provider == 'alipay' and acquirer.alipay_payment_method == 'express_checkout':  # TODO ARJ bool(acquirer) was useless here and won't be necessary anymore
-            currency_id = int(vals.get('currency_id'))
-            if currency_id:
-                currency = self.env['res.currency'].sudo().browse(currency_id)
-                if currency and currency.name != 'CNY':  # TODO ARJ same as for acquirer
-                    _logger.info("Only CNY currency is allowed for Alipay Express Checkout")
-                    # TODO ARJ triple quotes shouldn't be mixed with \n and are bad practice in general for translatable strings
-                    raise ValidationError(_("""
-                        Only transactions in Chinese Yuan (CNY) are allowed for Alipay Express Checkout.\n
-                        If you wish to use another currency than CNY for your transactions, switch your
-                        configuration to a Cross-border account on the Alipay payment acquirer in Odoo.
-                    """))
-        return True  # TODO ARJ this doesn't hurt but it's never evaluated. A check that returns None or raise is fine
+    # === CRUD METHODS ===#
+    @api.model
+    def create(self, vals):
```

Comment:
If you want some of VFE's chips, you better flag this with `@api.model_create_multi` and expect a list of vals.
If not, no big deal.